src/uu/install/src/install.rs

```diff
@@ -1042,3 +1043,23 @@ fn need_copy(from: &Path, to: &Path, b: &Behavior) -> UResult<bool> {
 
     Ok(false)
 }
+
+#[cfg(feature = "selinux")]
+fn set_selinux_context(path: &Path, behavior: &Behavior) -> UResult<()> {
+    // Skip if neither option is set
+    if !behavior.preserve_context && behavior.context.is_none() {
+        Ok(())
+    } else if behavior.preserve_context {
+        // If preserve_context is set, get the context from the source file
+        // and set it on the target file
+        // We would implement this when copying files in the copy() function
+        // This is handled in the copy() function below
+        Ok(())
+    } else if let Some(ref context) = behavior.context {
+        // Use the provided context set by -Z/--context
+        set_selinux_security_context(path, Some(context))
+            .map_err(|e| InstallError::SelinuxContextFailed(e.to_string()).into())
+    } else {
+        Ok(())
+    }
```

Comment:
I think you can simplify the function to something like:
```suggestion
    if let Some(ref context) = behavior.context && !behavior.preserve_context {
        // Use the provided context set by -Z/--context
        set_selinux_security_context(path, Some(context))
            .map_err(|e| InstallError::SelinuxContextFailed(e.to_string()))?
    }
    Ok(())
```